chainer/training/triggers/interval.py

```diff
@@ -40,9 +42,16 @@ def __call__(self, trainer):
         """
         updater = trainer.updater
         if self.unit == 'epoch':
-            prev = self.count
-            self.count = updater.epoch_detail // self.period
-            return prev != self.count
+            epoch_detail = updater.epoch_detail
+            previous_epoch_detail = updater.previous_epoch_detail
```

Comment:
@delta2323 Thank you for sharing the discussion.

> IntervalTrigger cannot rely on Updater.previous_epoch_detail if it is not called at every epoch.

I agree this (maybe `every epoch` -> `every iteration`). Current design assumes that triggers are called for each iteration.

> We encountered this problem when we use IntervalTriggers to collect the statistics of data nodes in the computational graph. We associate a trigger for each data node in the computational graph.

What is `node(s)`? Could you share me some examples? 

> Actually it is true of triggers for controlling the invocation of extensions, that are fed with trigger option of the Trainer.extend method. But this assumption restricts the usage of triggers. 

So will you change the arguments of `trigger.__call__()`? Currently, it takes a trainer. I think this is specific. 

Even if we save `previous_epoch_detail` in triggers, the issue, how to treat the cases that updaters/triggers were serialized by older version and do not have `previous_epoch_detail` in it (https://github.com/chainer/chainer/pull/2484#discussion_r112811180). How do you think of it.